peps/pep-0734.rst

```diff
@@ -0,0 +1,800 @@
+
+      It refers only to if there is a thread
+      running a script (code) in the interpreter's ``__main__`` module.
+      That basically means whether or not ``Interpreter.exec()`` is
+      running in some thread.  Code running in sub-threads is ignored.
+
+* ``prepare___main__(**kwargs)``
+      Bind one or more objects in the interpreter's ``__main__`` module.
+
+      The keyword argument names will be used as the attribute names.
+      The values will be bound as new objects, though exactly equivalent
+      to the original.  Only objects specifically supported for passing
+      between interpreters are allowed.  See `Shareable Objects`_.
+
+      ``prepare___main__()`` is helpful for initializing the
+      globals for an interpreter before running code in it.
+
+* ``exec(code, /)``
+      Run the given source code in the interpreter
+      (in the current thread).
+
+      This is essentially equivalent to switching to this interpreter
+      in the current thread and then calling the builtin ``exec()``
+      using this interpreter's ``__main__`` module's ``__dict__`` as
+      the globals and locals.
+
+      ``exec()`` does not reset the interpreter's state nor
+      the ``__main__`` module, neither before nor after, so each
+      successive call picks up where the last one left off.  This can
+      be useful for running some code to initialize an interpreter
+      (e.g. with imports) before later performing some repeated task.
+
+Comparison with builtins.exec()
+-------------------------------
+
+``Interpreter.exec()`` is essentially the same as the builtin
+``exec()``, except it targets a different interpreter, using that
+interpreter's distinct runtime state.
+
+Here are the relevant characteristics of the builtin ``exec()``,
+for comparison:
+
+* It runs in the current OS thread and pauses whatever was running there,
+  which resumes when ``exec()`` finishes.  No other threads are affected.
+  (To avoid pausing the current thread, run ``exec()``
+  in a ``threading.Thread``.)
+* It executes against a namespace, by default the ``__dict__`` of the
+  current module (i.e. ``globals()``).  It uses that namespace as-is
+  and does not clear it before or after.
+* When code is run from the command-line (e.g. ``-m``) or the REPL,
+  the "current" module is always the ``__main__`` module
+  of the target (main) interpreter.
+* It discards any object returned from the executed code::
+
+   def func():
+       global spam
+       spam = True
+       return 'a returned value'
+
+   ns = {}
+   res = exec(func.__code__, ns)
+   # Nothing is returned.  The returned string was discarded.
+   assert res is None, res
+   assert ns['spam'] is True, ns
+
+* It propagates any uncaught exception from the code it ran.
+  The exception is raised from the ``exec()`` call in the thread
+  that originally called ``exec()``.
+
+The only difference is that, rather than propagate the uncaught
```

Comment:
This paragraph seems logically to be part of the last bullet. Why not indent it accordingly?

Reply:
I'm moving the paragraph elsewhere.